apps/data-feeds-config-generator/src/generation/update/crypto-providers.ts

```diff
@@ -0,0 +1,99 @@
+import { Schema as S } from 'effect';
+
+import {
+  cryptoPriceFeedsArgsSchema,
+  NewFeedsConfig,
+  readConfig,
+} from '@blocksense/config-types';
+
+import {
+  getAllProvidersForPair,
+  getExchangesPriceDataForPair,
+  removePriceOutliers,
+} from '../../data-services/processors/crypto-providers/data-transformers';
+import { detectPriceOutliers } from '../../data-services/processors/crypto-providers/outliers-detector';
+import { CryptoProviderData } from '../../data-services/processors/crypto-providers/types';
+import { SimplifiedFeed } from '../initial/types';
+import { getCryptoProvidersData } from '../../data-services/processors/crypto-providers/data-collection';
+
+/**
+ * Updates the configuration for exchange arguments in the feeds configuration.
+ *
+ * This function retrieves the current feeds configuration, fetches crypto providers data,
+ * normalizes the providers data, and updates the feeds configuration with the appropriate
+ * exchange arguments for each feed. It ensures that only feeds with the oracle ID
+ * 'crypto-price-feeds' and containing 'aggregators' in their arguments are updated.
+ *
+ */
+export async function updateExchangesArgumentConfig(): Promise<NewFeedsConfig> {
+  const configFileName = 'feeds_config_v2';
+  let feedsConfig = await readConfig(configFileName);
+
+  const providersData = await getCryptoProvidersData();
+
+  const normalizedProvidersData = normalizeAllPriceFeedsProviders(
+    feedsConfig.feeds,
+    providersData,
+  );
```

Comment:
I think the code is more readable when the single-use functions are inlined:

```suggestion
  let normalizedProvidersData: CryptoProviderData[] = providersData;
  for (const feed of feeds) {
    const pair = feed.additional_feed_info.pair;
    const initialExchangePrices = getExchangesPriceDataForPair(
      pair,
      providersData,
    );
    const outlierExchanges = detectPriceOutliers(
      initialExchangePrices,
      ({ dataSourceName, price }) => 
        console.log(
          `Detected Outlier : ${dataSourceName} (${price}) ` +
          `for feed ${feed.full_name}`,
        )
    );
  
    normalizedProvidersData = removePriceOutliers(
      providersData,
      pair,
      outlierExchanges,
    );
  }
```